daemon/cmd/endpoint.go

```diff
@@ -176,6 +176,7 @@ func (d *Daemon) fetchK8sLabelsAndAnnotations(nsName, podName string) (*slim_cor
 
 func invalidDataError(ep *endpoint.Endpoint, err error) (*endpoint.Endpoint, int, error) {
 	ep.Logger(daemonSubsys).WithError(err).Warning("Creation of endpoint failed due to invalid data")
+	ep.SetState(endpoint.StateInvalid, "Invalid endpoint")
```

Comment:
I checked through as well to see that we don't need this anywhere else such as `errorDuringCreation()`, but that path is fine because it invokes `deleteEndpointQuiet()` which would transition the endpoint into `StateDisconnecting`.